lib/basevm.py

```diff
@@ -36,7 +37,9 @@ def param_clear(self, param_name):
                      param_name)
 
     def name(self):
-        return self.param_get('name-label')
+        n = self.param_get('name-label')
+        assert isinstance(n, str)
+        return n
```

Comment:
Shouldn't `param_get` instead be declared to return `str`?

Reply:
Possibly, but I'm not sure, so I only changed things where I was confident enough!

Reply:
`param_get` should really be declared as returning `Optional[str]`, and `name-label` is a non-optional field in the XAPI DB, so that indeed makes sense.  But it will be quite some work if we want to handle all mandatory field getters this way.

We could have a generic `param_get_mandatory_str` to factor in the behavior and typing, but the way param_get is defined (on every class) still makes it too heavy. We'd have to get to the refactor of those I've been thinking about for some time, but not in this R :sweat_smile: